doc/sphinx/architecture.md

```diff
@@ -0,0 +1,252 @@
+# Overview
+This document describes the high-level architecture of *Mir*, including a diagram and a brief introduction of the libraries.
+
+# Audience
+This document is intended to provide contributors to *Mir* an overview of *Mir*'s systems. It is *not* intended to guide compositor-implementers, unless they are specifically interested in the internals of *Mir*.
+
+# Diagram
+![Architecture Diagram](./architecture.png "Architecture Diagram")
+
+# Top-Level Folders
+## `/include`
+- The `include` folder at the root of the project provides the public-facing API.
+- This directory *strictly* includes the interfaces that are useful to a compositor-implementer.
+- If one is only interested in implementing their own *Mir*-based compositor, they need not look any further than this directory.
+
+## `/src`
+- The `src` folder at the root of the project comprises the core implementation of *Mir*.
+
+## `/examples`
+- A collection of demo *Mir* compositors that are used as examples of how one might build a true compositor. Examples include:
+  - `miral-app`
+  - `miral-shel`
+  - `mir_demo_server`
+
+## `/tests`
+- Unit tests for the entire project.
+
+# Libraries
+The following is the library tree:
+```
+├── miral
+├── miroil
+├── mircore
+├── mircommon
+├── mircookie
+└── mirserver
+    ├── mircompositor
+    ├── mirinput
+    ├── mirfrontend
+    ├── mirfrontend-wayland
+    ├── mirfrontend-xwayland
+    ├── mirshell
+    ├── mirshelldecoration
+    ├── mirscene
+    ├── mirgraphics
+    ├── mirreport
+    ├── mirlttng
+    ├── mirlogging
+    ├── mirnullreport
+    ├── mirconsole
+    ├── mirgl
+    └── mirrenderergl
+├── mirplatform
+├── mirwayland
+├── graphics-gbm-kms
+├── graphics-eglstreamkms
+├── graphics-x11
+├── graphics-wayland
+├── input-evdev
+└── renderer-egl-generic
+```
+
+## miral
+- The "Mir Abstraction Layer" (aka `MirAL` or `miral`) is an API that makes Mir easy for compositor-implementers to work with. It provides the core window management concepts and an interface that is more ABI stable than Mir's internal mirserver API.
+- Public-facing
+- Directories:
+  - `include/miral`
+  - `src/miral`
+  
+## miroil
+- A custom *Mir* library for the [Lomiri](https://lomiri.com/) folks
+- This is unimportant for 99% of people
+- Directories:
+  - `include/miroil`
+  - `src/miroil`
+
+## mircore
+- Fundamental data structures like 2D points, Rectangles, File Descriptor, etc.
+- Public-facing
+- Directories:
+  - `include/core`
+  - `src/core`
+  -
+## mircommon
+- Data structures and utility functions like Mir-event building, logging, keymaps, etc.
+- Differs from `mircore` in that the concepts map strictly onto *Mir*'s concepts instead of being general concepts like a Rectangle
+- Public-facing
+- Directories:
+  - `include/common`
+  - `src/common`
+  - `src/include/common`
+
+## mircookie
+- Provides event timestamps for inputs events that are difficult to spoof.
```

Comment:
@RAOF I suspect that mircookie isn't actually useful now we're no longer shipping Mir's input events to clients. WDYT?